app.rb

```diff
@@ -0,0 +1,127 @@
+require './student'
+require './teacher'
+require './book'
+require './rental'
+require './capitalize_decorator'
+class App
+  def initialize
+    @user_input = 0
+    @book_list = []
+    @people = []
+    @rental_list = []
+  end
+
+  def run
+    loop do
+      print "\nPlease choose an option by entering a number:\n1 - List all books\n2 - List all people\n3 - Create a person\n4 - Create a book\n5 - Create a rental\n6 - List all rentals for a given person id\n7 - Exit\n"
+      @user_input = gets.chomp.to_i
+      (1..7).include?(@user_input) ? switch_case : wrong_input
+    end
+  end
+
+  def switch_case
+    case @user_input
+    when 1 then list_all_books
+    when 2 then list_all_people
+    when 3 then create_person
+    when 4 then create_book
+    when 5 then create_rental
+    when 6 then list_all_rental
+    when 7 then exit_app
+    end
+  end
+
+  def wrong_input
+    puts "Please Enter a number in the menu\n"
+  end
+
+  def list_all_books
+    if @book_list.empty?
+      puts "\nThere are no books here"
+    else
+      @book_list.each { |book| puts %(Title: "#{book.title}", Author: #{book.author}) }
+    end
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts "\nThere are no people here"
+    else
+      @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    end
+  end
+
+  def exit_app
+    puts 'Thank you for using this app'
+    exit
+  end
+
+  def create_person
+    puts 'Do you want to create a student(1) or a teacher(2)? [Input The Number]:'
+    @user_input = gets.chomp.to_i
+    print 'Age: '
+    age = gets.chomp.to_i
+    # check age not to have wierd values
+    print 'Name: '
+    name = gets.chomp
+    case @user_input
+    when 1
+      print 'Has parent permission? [Y | N] '
+      parent_permission = gets.chomp.downcase
+      parent_permission = parent_permission == 'y'
+      # use decorator to capitalize name
+      @people << Student.new('none', age, name, parent_permission)
+    when 2
+      print 'Specialization: '
+      specialization = gets.chomp
+      parent_permission = true
+      # use decorator to capitalize name
+      @people << Teacher.new(age, name, parent_permission, specialization)
+    end
+    puts '\nPerson created successfully'
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    # capitalize both title and author
```

Comment:
[OPTIONAL] To maintain clean code, it would be preferable to remove the comments once the work is complete.